4_params_munge/src/munge_bathy.R

```diff
@@ -101,3 +101,81 @@ munge_mndow_bathy <- function(out_ind, bathy_zip_ind, mndow_xwalk_ind){
     saveRDS(data_file)
   gd_put(out_ind, data_file)
 }
+
+munge_ndgf_bathy <- function(out_ind, ndgf_contour_ind, ndgf_xwalk_ind, ndgf_surface_sf_ind, crs){
+
+  ndgf_xwalk <- readRDS(sc_retrieve(ndgf_xwalk_ind))
+
+  # Now crosswalk with NHD lake ids
+  ndgf_contours <- readRDS(sc_retrieve(ndgf_contour_ind)) %>%
+    mutate(NDGF_ID = sprintf("ndgf_%s", LAKE)) %>%
+    inner_join(ndgf_xwalk, by = 'NDGF_ID') %>%
+    rename(other_ID = NDGF_ID,
+           depths = CONTOUR) %>%
+    dplyr::select(site_id, other_ID, depths)
+
+  # Due to some issues with incomplete surface polygons in the contour data
+  #   for some lakes causing zero area lakes and ultimately monotonicity issues,
+  #   replace all contour shapes (depth = 0) with the polygons from ndgf_lakesvw.
+
+  # We will treat these separately until after issues with some of the contour data are resolved
+
+  # Many of the lakes have multiple polygons at the same depth. Some of the smaller
+  # polygons at the same depth are contained within a larger polygon at the same depth.
+  # This doesn't make much sense and we end up double counting the area. So, union them
+  # first to reconcile internal boundaries by lake and then find the area.
+
+  ndgf_non_surface_geoms <- ndgf_contours %>%
+    filter(depths != 0) %>%
+    group_by(site_id, other_ID, depths) %>%
+    summarize(geometry = st_union(geometry)) %>%
```

Comment:
The [docs](https://r-spatial.github.io/sf/reference/geos_combine.html) don't make it super clear that this does exactly what we want. Have you checked by plotting? In the case you are talking about, the separate polygons should stay separate, but the overlapping ones should merge into one. Is that what is happening?